apps/website/src/components/ComponentIcon.astro

```diff
@@ -0,0 +1,102 @@
+---
+import { renderJSX } from 'astro/dist/runtime/server';
+import fs from 'node:fs';
+
+export type Props = {
+  /** name of the file to read code example from, should be located in ~/examples folder */
+  icon: string;
+};
+
+const { icon } = Astro.props;
+
+// baseUrl refers to project root
+const baseUrl = import.meta.env.PROD
+  ? new URL('..', import.meta.url).href
+  : new URL('../../', import.meta.url);
+
+let code = '';
+
+if (icon) {
+  const codePath = new URL(`./src/icons/component-icons/${icon}.astro`, baseUrl);
+  code = await fs.promises.readFile(codePath, 'utf8');
+}
+
+// const componentName = icon.split('.')[0];
+// const capitalize = (s: string) => s.charAt(0).toUpperCase() + s.slice(1);
+---
+
+<div class='thumbnail' set:html={code} />
+
+<style lang='scss' is:global>
+  .thumbnail svg {
```

Comment:
If these styles are global, we need to come up with a way to scope them. I can think of two approaches to this:
1. move most of the styles into the thumbnails and scope them there. then expose a few custom properties to control them from outside.
2. create a scoped root class name, and then use `:global` to target the classes inside the svg. this is similar to what i'm doing with the stuff inside content-wrapper in `_layout.astro`.

Reply:
Should be good now.